components/private_channel/rust/ffi/Makefile

```diff
@@ -0,0 +1,41 @@
+CFLAGS :=
+
+ifdef NDEBUG
+CFLAGS += -DNDEBUG=${NDEBUG}
+endif
+
+ifdef NO_CXXEXCEPTIONS
+CFLAGS += -DNO_CXXEXCEPTIONS=${NO_CXXEXCEPTIONS}
+endif
+
+all: examples/cpp.out
+
+src/lib.h: src/lib.rs
+	cbindgen -o src/lib.h
+
+sample: examples/cpp.out
+	./examples/cpp.out
```

Comment:
Agreed and as the likely originator of this convention - I'm sorry! I swear the original actually started as an example before it morphed into a sort of FFI test suite :smile:

Reply:
- [ ] rename from `examples/` -> `test/`

Reply:
- [ ] include FFI tests as part of tests run in CI pipeline

Reply:
> - [ ] include FFI tests as part of tests run in CI pipeline

@gpestana As long as it's clear how to test whenever changes are made here, I didn't intend to let this hold up getting the feature merged. To my knowledge we don't have this on any FFI components; it would be great to have for all of those, but it will likely require broader architectural discussion with @bridiver and others.

Reply:
I agree. I will open an issue here for including FFI tests as part of the CI pipeline